attest/activation.go

```diff
@@ -88,32 +98,68 @@ func (p *ActivationParameters) checkTPM12AKParameters() error {
 	return nil
 }
 
+// VerifyOpts specifies options passed to (*AttestationParameters).Verify()
+type VerifyOpts struct {
+	// SelfAttested set to true ensures that the attestation is self-signed,
+	// set to false ensures that the attestation is not self-signed.
+	SelfAttested bool
+	// Restricted set to true ensures that the verified key was created by TPM
```

Comment:
If we're going to support this, please add a note that the SK won't work as a crypto.Signer if it's restricted.

Also do we actually need this for something? Maybe just drop it entirely?

Reply:
For AK's AttestationParameters this check is required. I'll add a note.

Reply:
Same as my other comment, this probably shouldn't be exposed to the user then. We can't require a new call since it's not required today.

Reply:
Where a new call would be required?

But for hiding options from callers, what about renaming it to `verify(...)` and then expose `VerifyAppKey(pub)` that would call `verify(...)` with Restricted=false?